pkg/cloudprovider/plugins/internal/accounts_common.go

```diff
@@ -137,9 +142,13 @@ func (c *cloudCommon) updateCloudAccountConfig(client client.Client, credentials
 		return err
 	}
 
-	if err := currentConfig.serviceConfig.UpdateServiceConfig(serviceConfig); err != nil {
-		cloudAccountConfigState = false
-		return err
+	if currentConfig.serviceConfig == nil {
+		currentConfig.serviceConfig = serviceConfig
+	} else {
```

Comment:
Consider move the update outside of the else block and just return nil after L146 in the if block.

Reply:
We want to continue if no error occurs. For example, if someone adds some code at line 153, that won't be executed.